File: apps/android/feature/ai/data/src/main/java/com/scribblefit/feature/ai/data/repository/TelemetryRepositoryImpl.kt
package com.scribblefit.feature.ai.data.repository

import com.scribblefit.core.network.ScribbleFitApi
import com.scribblefit.core.network.model.TelemetryRequest
import com.scribblefit.feature.ai.domain.engine.TelemetryRepository
import com.scribblefit.feature.ai.domain.model.TelemetryData
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class TelemetryRepositoryImpl @Inject constructor(
    private val api: ScribbleFitApi
) : TelemetryRepository {

    override suspend fun reportError(data: TelemetryData): Result<Unit> = runCatching {
        api.reportError(
            TelemetryRequest(
                rawText = data.rawText,
                promptVersion = data.promptVersion,
                errorMessage = data.errorMessage,
                errorCode = data.errorCode,
                deviceModel = data.deviceModel
            )
        )
        Unit
    }
}

File: apps/ios/ScribbleFit/Sources/Features/AI/Data/Repository/TelemetryRepositoryImpl.swift
import Foundation

public final class TelemetryRepositoryImpl: TelemetryRepository {
    private let networkClient: ScribbleFitNetworkClient
    
    public init(networkClient: ScribbleFitNetworkClient = .shared) {
        self.networkClient = networkClient
    }
    
    public func reportError(data: TelemetryData) async throws {
        let request = TelemetryRequest(
            rawText: data.rawText,
            promptVersion: data.promptVersion,
            errorMessage: data.errorMessage,
            errorCode: data.errorCode,
            deviceModel: data.deviceModel
        )
        try await networkClient.reportError(request: request)
    }
}
